torch/csrc/autograd/generated/python_torch_functions.cpp

```diff
@@ -2606,7 +2607,7 @@ static PyObject * THPVariable_bartlett_window(PyObject* self_, PyObject* args, P
         .layout(r.layout(2).layout)
         .requires_grad(r.toBool(5))
         .pinned_memory(r.toBool(4));
-    return wrap(dispatch_bartlett_window(window_length, options));
+    return wrap(dispatch_bartlett_window(window_length, r.scalartypeOptional(1), c10::optional<at::Layout>(r.layout(2).layout), r.deviceOptional(3), r.toBoolOptional(4)));
```

Comment:
Github won't let me comment where I want to, so I'll comment here. The PythonArgParser says: "bartlett_window(int64_t window_length, *, ScalarType dtype=None, Layout layout=torch.strided, Device device=None, bool pin_memory=False, bool requires_grad=False)"

However, the native_functions.yaml string says: 
```
- func: bartlett_window(int window_length, *, ScalarType? dtype=None, Layout? layout=None, Device? devi
ce=None, bool? pin_memory=None) -> Tensor
```
I suspect the `Layout layout = torch.strided` in the PythonArgParser is special cased in the codegen and it might need to change.

Reply:
All of the TensorOptions have their default values in case when they are not specified. You have see an example [here](https://github.com/pytorch/pytorch/blob/master/tools/autograd/templates/python_torch_functions.cpp#L191-L194). As of right now i would stick to existing logic but, in case we disagree with this logic, we can change it in the future